services/core-api/app/api/now_applications/resources/administrative_amendment_list_resource.py

```diff
@@ -0,0 +1,180 @@
+from app.api.now_applications.models.now_party_appointment import NOWPartyAppointment
+from app.api.mines.documents.models.mine_document import MineDocument
+from app.api.now_applications.models.now_application_document_xref import NOWApplicationDocumentXref
+from app.api.now_applications.models.administrative_amendments.application_reason_code_xref import ApplicationReasonXref
+from app.api.now_applications.response_models import NOW_APPLICATION_MODEL
+from app.api.now_applications.models.applications_view import ApplicationsView
+from app.api.utils.access_decorators import requires_role_edit_permit, requires_any_of, VIEW_ALL, GIS
+from app.api.utils.resources_mixins import UserMixin
+from app.api.utils.custom_reqparser import CustomReqparser
+
+
+def map_notice_of_work_type_from_permit_number(permit_number):
+    first_charecter = permit_number[0]
+    return {'P': 'PLA', 'G': 'SAG', 'M': 'MIN', 'C': 'COL', 'Q': 'QCA'}.get(first_charecter, None)
+
+
+class AdministrativeAmendmentListResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument('mine_guid', type=str, required=True)
+    parser.add_argument('received_date', type=str, required=True)
+    parser.add_argument('permit_amendment_guid', type=str, required=True)
+    parser.add_argument('permit_id', type=int, required=True)
+    parser.add_argument('application_source_type_code', type=str, required=True)
+    parser.add_argument('application_reason_codes', type=list, location='json', required=True)
+
+    @api.doc(description='Adds a Notice of Work to a mine/permit.', params={})
+    @requires_role_edit_permit
+    @api.marshal_with(NOW_APPLICATION_MODEL, code=201)
+    def post(self):
+        data = self.parser.parse_args()
+        current_app.logger.debug("this is administrative amendment post endpoint")
+
+        mine = Mine.find_by_mine_guid(data['mine_guid'])
+        permit = Permit.find_by_permit_id(data['permit_id'])
+        permit_amendment = PermitAmendment.find_by_permit_amendment_guid(
+            data['permit_amendment_guid'])
+        has_existing_administrative_amendments = ApplicationsView.query.filter_by(
+            source_permit_amendment_guid=permit_amendment.permit_amendment_guid,
+            application_type_code='ADA').filter(
+                ApplicationsView.now_application_status_code.in_(['AIA', 'REJ'])).count() > 0
+
+        application = None
+        if permit_amendment.now_application_guid:
+            application = NOWApplication.find_by_application_guid(
+                permit_amendment.now_application_guid)
+
+        err_str = ''
+        if not mine:
+            err_str += 'Mine not found. '
+        if not permit:
+            err_str += 'Permit not found. '
+        if not permit_amendment:
+            err_str += 'Permit amendment not found '
+        if has_existing_administrative_amendments:
+            err_str += 'You cannot have multiple in progress administrative amendments '
```

Comment:
Can we make the punctuation consistent with these error messages?
Also, need a hyphen: "You cannot have multiple in-progress administrative amendments"

Reply:
done